f8a_worker/workers/repo_dependency_finder.py

```diff
@@ -0,0 +1,160 @@
+import requests
+import traceback
+
+from f8a_worker.base import BaseTask
+from f8a_worker.errors import TaskError
+from f8a_worker.utils import GREMLIN_SERVER_URL_REST
+from f8a_worker.workers.mercator import MercatorTask
+from f8a_worker.workers.dependency_parser import GithubDependencyTreeTask
+
+
+class RepoDependencyFinderTask(BaseTask):
+    """Finds out direct and indirect dependencies from a given github repository."""
+
+    _mercator = MercatorTask.create_test_instance(task_name='RepoDependencyFinderTask')
+
+    def execute(self, arguments=None):
+        """Task code.
+
+        :param arguments: dictionary with task arguments
+        :return: {}, results
+        """
+        self._strict_assert(arguments.get('service_token'))
+
+        github_repo = arguments.get('github_repo')
+        dependencies = []
+        repo_cves = []
+        if arguments.get('epv_list') in (None, []):
+            # self._strict_assert(arguments.get('epv_list'))
+            if len(arguments.get('epv_list')) > 0:
+                for epv in arguments.get('epv_list'):
+                    dependencies.append('{ecosystem}:{package}:{version}'
+                                        .format(ecosystem=epv.get('ecosystem'),
+                                                package=epv.get('name'),
+                                                version=epv.get('version')))
+            try:
+                repo_cves = self.get_cve(dependencies)
+            except TaskError as e:
+                raise TaskError('Failed to get CVEs')
+        else:
+            dependencies = list(GithubDependencyTreeTask.extract_dependencies(github_repo))
+            try:
+                # forward only the available dependencies in the system. Unknown
+                # dependencies are not going to be ingested for osioUserNotificationFlow.
+                repo_cves = self.create_repo_node_and_get_cve(github_repo, dependencies)
+            except TaskError as e:
+                raise TaskError('Failed to Create Repo Node')
+
+        report = self.generate_report(repo_cves=repo_cves, deps_list=dependencies)
+        return report
+
+    def create_repo_node_and_get_cve(self, deps_list):
+        """Create a repository node in the graphdb and create its edges to all deps.
+
+        :param github_repo:
+        :param dependencies:
+        :return:
+        """
+        gremlin_str = "repo=g.V().has('repo_url', '{repo_url}').tryNext().orElseGet{{" \
+                      "graph.addVertex('vertex_label', 'Repo', 'repo_url', '{repo_url}')}};" \
+                      "g.V(repo).outE('has_dependency').drop().iterate();".format(
+                          repo_url=github_repo)
+
+        for pkg in deps_list:
+            ecosystem = pkg.split(':')[0]
+            version = pkg.split(':')[-1]
+            name = pkg.replace(ecosystem + ':', '').replace(':' + version, '')
+            gremlin_str += "ver=g.V().has('pecosystem', '{ecosystem}').has('pname', '{name}')." \
```

Comment:
Same here, the concatenation magic can save you a lot of "\" typing :slightly_smiling_face: